---
title: "0129-Final version- with ADT and RNA integration"
output:
  html_document:
    df_print: paged
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

load all packages for the downstream analysis

```{r packages}
library('Seurat')
library(uwot)
library(dplyr)
library(Matrix)
library(cowplot)
library(data.table)
library(magrittr)
library(tidyverse)
library(RCurl)
library(purrr)
library(pheatmap)
library(MAST)
library(RColorBrewer)
library(grid)
library(gtable)
library(scds)
library(scater)
library(rsvd)
library(Rtsne)
library(cowplot)
library(SingleCellExperiment)
library(scDblFinder)
library(ggplot2)
library(harmony)
library(clustree)
library(Seurat)
library(SeuratData)
library(cowplot)
library(dplyr)
```

## load function code for RNA and ADT name trim

function to trim ADT:

```{r pressure}
trim.feature.names <- function(inmat){
  newnames <- sapply(strsplit(rownames(inmat), "_"),
                     function(x) {
                       if(length(x)==1) return(x)
                       else return(paste(x[-length(x)], collapse="_"))
                     })
  rownames(inmat) <- newnames
  return(inmat)
}
```

function to trim RNA:
```{r}
trim.feature.names.RNA <- function(inmat){
newnames <- sapply(strsplit(rownames(inmat), "_"),
function(x) {
if(length(x)==1) return(x)
else return(paste(x[length(x)], collapse="_"))
})
rownames(inmat) <- newnames
return(inmat)
}
```
## load all the AD, Pso, and AE samples

```{r}
####### Normal samples
Skin150.data <- Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/150")
Skin154.data <- Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/154")
Skin155.data <- Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/155")
Skin169.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/169", gene.column=2))
Skin195.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/195", gene.column=2))
Skin204.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/204", gene.column=2))
Skin207.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/207", gene.column=2))

####### psoriasis samples
Skin165.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/165", gene.column=2))
Skin173.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/173", gene.column=2))
Skin194.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/194", gene.column=2))
Skin199.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/199", gene.column=2))
Skin211.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/211", gene.column=2))
Skin222.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/222", gene.column=2))
Skin234.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/234", gene.column=2))
Skin235.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/235", gene.column=2))

######### AD samples
Skin170.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/170", gene.column=2))
Skin198.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/198", gene.column=2))
Skin230.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/230", gene.column=2))
Skin231.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/231", gene.column=2))
Skin232.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/232", gene.column=2))
Skin233.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/233", gene.column=2))
Skin236.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/236", gene.column=2))

######### AE samples
Skin167.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/167", gene.column=2))
Skin174.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/174", gene.column=2))
Skin192.data <- (Read10X_h5("E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/192/Skin 192  Atypical Eczema-2.h5" , use.names = TRUE, unique.features = TRUE))
Skin200.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/200", gene.column=2))
Skin202.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/202", gene.column=2))
Skin218.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/218", gene.column=2))
Skin219.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/219", gene.column=2))

######### other samples
Skin141.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/141", gene.column=2))

Skin163.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/163", gene.column=2))

Skin175.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/175", gene.column=2))

Skin203.data <- (Read10X(data.dir = "E:/Human - 10 clusters/yale-method/1124-AD and PV data matrix/203", gene.column=2))
```

The following samples have mouse cells spike-in, we will keep 200 genes for comparision. We keep 200 is also for find doublets, as this function need the min(colSums(counts(object)))>0. If we use 100, the Skin231 will ==0.
```{r}
####### Normal samples
Skin195.data[["Gene Expression"]] <- CollapseSpeciesExpressionMatrix(Skin195.data[["Gene Expression"]], prefix = "GRCh38_", controls = "mm10___", ncontrols = 200)
Skin204.data[["Gene Expression"]] <- CollapseSpeciesExpressionMatrix(Skin204.data[["Gene Expression"]], prefix = "GRCh38_", controls = "mm10___", ncontrols = 200)
Skin207.data[["Gene Expression"]] <- CollapseSpeciesExpressionMatrix(Skin207.data[["Gene Expression"]], prefix = "GRCh38_", controls = "mm10___", ncontrols = 200)

####### psoriasis samples
Skin199.data[["Gene Expression"]] <- CollapseSpeciesExpressionMatrix(Skin199.data[["Gene Expression"]], prefix = "GRCh38_", controls = "mm10___", ncontrols = 200)
Skin211.data[["Gene Expression"]] <- CollapseSpeciesExpressionMatrix(Skin211.data[["Gene Expression"]], prefix = "GRCh38_", controls = "mm10___", ncontrols = 200)
Skin222.data[["Gene Expression"]] <- CollapseSpeciesExpressionMatrix(Skin222.data[["Gene Expression"]], prefix = "GRCh38_", controls = "mm10___", ncontrols = 200)
Skin234.data[["Gene Expression"]] <- CollapseSpeciesExpressionMatrix(Skin234.data[["Gene Expression"]], prefix = "GRCh38_", controls = "mm10___", ncontrols = 200)
Skin235.data[["Gene Expression"]] <- CollapseSpeciesExpressionMatrix(Skin235.data[["Gene Expression"]], prefix = "GRCh38_", controls = "mm10___", ncontrols = 200)

######### AD samples
Skin198.data[["Gene Expression"]] <- CollapseSpeciesExpressionMatrix(Skin198.data[["Gene Expression"]], prefix = "GRCh38_", controls = "mm10___", ncontrols = 200)
Skin230.data[["Gene Expression"]] <- CollapseSpeciesExpressionMatrix(Skin230.data[["Gene Expression"]], prefix = "GRCh38_", controls = "mm10___", ncontrols = 200)
Skin231.data[["Gene Expression"]] <- CollapseSpeciesExpressionMatrix(Skin231.data[["Gene Expression"]], prefix = "GRCh38_", controls = "mm10___", ncontrols = 200)
Skin232.data[["Gene Expression"]] <- CollapseSpeciesExpressionMatrix(Skin232.data[["Gene Expression"]], prefix = "GRCh38_", controls = "mm10___", ncontrols = 200)
Skin233.data[["Gene Expression"]] <- CollapseSpeciesExpressionMatrix(Skin233.data[["Gene Expression"]], prefix = "GRCh38_", controls = "mm10___", ncontrols = 200)
Skin236.data[["Gene Expression"]] <- CollapseSpeciesExpressionMatrix(Skin236.data[["Gene Expression"]], prefix = "GRCh38_", controls = "mm10___", ncontrols = 200)

######### AE samples
Skin200.data[["Gene Expression"]] <- CollapseSpeciesExpressionMatrix(Skin200.data[["Gene Expression"]], prefix = "GRCh38_", controls = "mm10___", ncontrols = 200)
Skin202.data[["Gene Expression"]] <- CollapseSpeciesExpressionMatrix(Skin202.data[["Gene Expression"]], prefix = "GRCh38_", controls = "mm10___", ncontrols = 200)
Skin218.data[["Gene Expression"]] <- CollapseSpeciesExpressionMatrix(Skin218.data[["Gene Expression"]], prefix = "GRCh38_", controls = "mm10___", ncontrols = 200)
Skin219.data[["Gene Expression"]] <- CollapseSpeciesExpressionMatrix(Skin219.data[["Gene Expression"]], prefix = "GRCh38_", controls = "mm10___", ncontrols = 200)

####
Skin203.data[["Gene Expression"]] <- CollapseSpeciesExpressionMatrix(Skin203.data[["Gene Expression"]], prefix = "GRCh38_", controls = "mm10___", ncontrols = 200)
```

Trim the ADT data

```{r}
rownames(x = Skin150.data[["Antibody Capture"]]) <- gsub(pattern = "_[control_]*TotalA", replacement = "", 
                                                          x = rownames(x = Skin150.data[["Antibody Capture"]]))
rownames(x = Skin154.data[["Antibody Capture"]]) <- gsub(pattern = "_[control_]*TotalA", replacement = "", 
                                                          x = rownames(x = Skin154.data[["Antibody Capture"]]))
rownames(x = Skin155.data[["Antibody Capture"]]) <- gsub(pattern = "_[control_]*TotalA", replacement = "", 
                                                          x = rownames(x = Skin155.data[["Antibody Capture"]]))


### Normal
Skin169.data$"Antibody Capture" <- trim.feature.names(Skin169.data$"Antibody Capture")
Skin195.data$"Antibody Capture" <- trim.feature.names(Skin195.data$"Antibody Capture")
Skin204.data$"Antibody Capture" <- trim.feature.names(Skin204.data$"Antibody Capture")
Skin207.data$"Antibody Capture" <- trim.feature.names(Skin207.data$"Antibody Capture")

### AD
Skin170.data$"Antibody Capture" <- trim.feature.names(Skin170.data$"Antibody Capture")
Skin198.data$"Antibody Capture" <- trim.feature.names(Skin198.data$"Antibody Capture")
Skin230.data$"Antibody Capture" <- trim.feature.names(Skin230.data$"Antibody Capture")
Skin231.data$"Antibody Capture" <- trim.feature.names(Skin231.data$"Antibody Capture")
Skin232.data$"Antibody Capture" <- trim.feature.names(Skin232.data$"Antibody Capture")
Skin233.data$"Antibody Capture" <- trim.feature.names(Skin233.data$"Antibody Capture")
Skin236.data$"Antibody Capture" <- trim.feature.names(Skin236.data$"Antibody Capture")

### Psoriasis
Skin165.data$"Antibody Capture" <- trim.feature.names(Skin165.data$"Antibody Capture")
Skin173.data$"Antibody Capture" <- trim.feature.names(Skin173.data$"Antibody Capture")
Skin194.data$"Antibody Capture" <- trim.feature.names(Skin194.data$"Antibody Capture")
Skin199.data$"Antibody Capture" <- trim.feature.names(Skin199.data$"Antibody Capture")
Skin211.data$"Antibody Capture" <- trim.feature.names(Skin211.data$"Antibody Capture")
Skin222.data$"Antibody Capture" <- trim.feature.names(Skin222.data$"Antibody Capture")
Skin234.data$"Antibody Capture" <- trim.feature.names(Skin234.data$"Antibody Capture")
Skin235.data$"Antibody Capture" <- trim.feature.names(Skin235.data$"Antibody Capture")


######### AE samples
Skin167.data$"Antibody Capture" <- trim.feature.names(Skin167.data$"Antibody Capture")
Skin192.data$"Antibody Capture" <- trim.feature.names(Skin192.data$"Antibody Capture")
Skin200.data$"Antibody Capture" <- trim.feature.names(Skin200.data$"Antibody Capture")
Skin202.data$"Antibody Capture" <- trim.feature.names(Skin202.data$"Antibody Capture")
Skin218.data$"Antibody Capture" <- trim.feature.names(Skin218.data$"Antibody Capture")
Skin219.data$"Antibody Capture" <- trim.feature.names(Skin219.data$"Antibody Capture")

###others
Skin163.data$"Antibody Capture" <- trim.feature.names(Skin163.data$"Antibody Capture")
Skin175.data$"Antibody Capture" <- trim.feature.names(Skin175.data$"Antibody Capture")
Skin203.data$"Antibody Capture" <- trim.feature.names(Skin203.data$"Antibody Capture")
```

Check the ADT name to make sure whether they are the same
```{r}
rownames(Skin150.data[["Antibody Capture"]])
rownames(Skin154.data[["Antibody Capture"]])
rownames(Skin155.data[["Antibody Capture"]])
rownames(Skin169.data[["Antibody Capture"]])
rownames(Skin195.data[["Antibody Capture"]])
rownames(Skin204.data[["Antibody Capture"]])
rownames(Skin207.data[["Antibody Capture"]])

### AD
rownames(Skin170.data[["Antibody Capture"]])
rownames(Skin198.data[["Antibody Capture"]])
rownames(Skin230.data[["Antibody Capture"]])
rownames(Skin231.data[["Antibody Capture"]])
rownames(Skin232.data[["Antibody Capture"]])
rownames(Skin233.data[["Antibody Capture"]])
rownames(Skin236.data[["Antibody Capture"]])

### Psoriasis
rownames(Skin165.data[["Antibody Capture"]])
rownames(Skin173.data[["Antibody Capture"]])
rownames(Skin194.data[["Antibody Capture"]])
rownames(Skin199.data[["Antibody Capture"]])
rownames(Skin211.data[["Antibody Capture"]])
rownames(Skin222.data[["Antibody Capture"]])
rownames(Skin234.data[["Antibody Capture"]])
rownames(Skin235.data[["Antibody Capture"]])


######### AE samples
rownames(Skin167.data[["Antibody Capture"]])
rownames(Skin192.data[["Antibody Capture"]])
rownames(Skin200.data[["Antibody Capture"]])
rownames(Skin202.data[["Antibody Capture"]])
rownames(Skin218.data[["Antibody Capture"]])
rownames(Skin219.data[["Antibody Capture"]])

######other
rownames(Skin163.data[["Antibody Capture"]])
rownames(Skin175.data[["Antibody Capture"]])
rownames(Skin203.data[["Antibody Capture"]])
```

Trim the object with different ADT names
```{r}
###Skin150
rownames(x = Skin150.data[["Antibody Capture"]]) <- gsub("CD141-Thrombomodulin","CD141", x = rownames(x = Skin150.data[["Antibody Capture"]]))
rownames(x = Skin150.data[["Antibody Capture"]]) <- gsub("CD294-CRTH2", "CD294", x = rownames(x = Skin150.data[["Antibody Capture"]]))
rownames(x = Skin150.data[["Antibody Capture"]]) <- gsub("CD127-IL-7R?", "CD127", x = rownames(x = Skin150.data[["Antibody Capture"]]))
rownames(x = Skin150.data[["Antibody Capture"]]) <- gsub("CD207-Langerin", "CD207", x = rownames(x = Skin150.data[["Antibody Capture"]]))
rownames(x = Skin150.data[["Antibody Capture"]]) <- gsub("CD197-CCR7", "CD197", x = rownames(x = Skin150.data[["Antibody Capture"]]))
rownames(x = Skin150.data[["Antibody Capture"]]) <- gsub("CD196-CCR6", "CD196", x = rownames(x = Skin150.data[["Antibody Capture"]]))
rownames(x = Skin150.data[["Antibody Capture"]]) <- gsub("CD195-CCR5", "CD195", x = rownames(x = Skin150.data[["Antibody Capture"]]))
rownames(x = Skin150.data[["Antibody Capture"]]) <- gsub("CD56-NCAM", "CD56", x = rownames(x = Skin150.data[["Antibody Capture"]]))
rownames(x = Skin150.data[["Antibody Capture"]]) <- gsub("[?]", "", x = rownames(x = Skin150.data[["Antibody Capture"]]))

### Skin154
rownames(x = Skin154.data[["Antibody Capture"]]) <- gsub("CD141-Thrombomodulin","CD141", x = rownames(x = Skin154.data[["Antibody Capture"]]))
rownames(x = Skin154.data[["Antibody Capture"]]) <- gsub("CD294-CRTH2", "CD294", x = rownames(x = Skin154.data[["Antibody Capture"]]))
rownames(x = Skin154.data[["Antibody Capture"]]) <- gsub("CD127-IL-7R?", "CD127", x = rownames(x = Skin154.data[["Antibody Capture"]]))
rownames(x = Skin154.data[["Antibody Capture"]]) <- gsub("CD207-Langerin", "CD207", x = rownames(x = Skin154.data[["Antibody Capture"]]))
rownames(x = Skin154.data[["Antibody Capture"]]) <- gsub("CD197-CCR7", "CD197", x = rownames(x = Skin154.data[["Antibody Capture"]]))
rownames(x = Skin154.data[["Antibody Capture"]]) <- gsub("CD196-CCR6", "CD196", x = rownames(x = Skin154.data[["Antibody Capture"]]))
rownames(x = Skin154.data[["Antibody Capture"]]) <- gsub("CD195-CCR5", "CD195", x = rownames(x = Skin154.data[["Antibody Capture"]]))
rownames(x = Skin154.data[["Antibody Capture"]]) <- gsub("CD56-NCAM", "CD56", x = rownames(x = Skin154.data[["Antibody Capture"]]))
rownames(x = Skin154.data[["Antibody Capture"]]) <- gsub("[?]", "", x = rownames(x = Skin154.data[["Antibody Capture"]]))

### Skin155
rownames(x = Skin155.data[["Antibody Capture"]]) <- gsub("CD141-Thrombomodulin","CD141", x = rownames(x = Skin155.data[["Antibody Capture"]]))
rownames(x = Skin155.data[["Antibody Capture"]]) <- gsub("CD294-CRTH2", "CD294", x = rownames(x = Skin155.data[["Antibody Capture"]]))
rownames(x = Skin155.data[["Antibody Capture"]]) <- gsub("CD127-IL-7R?", "CD127", x = rownames(x = Skin155.data[["Antibody Capture"]]))
rownames(x = Skin155.data[["Antibody Capture"]]) <- gsub("CD207-Langerin", "CD207", x = rownames(x = Skin155.data[["Antibody Capture"]]))
rownames(x = Skin155.data[["Antibody Capture"]]) <- gsub("CD197-CCR7", "CD197", x = rownames(x = Skin155.data[["Antibody Capture"]]))
rownames(x = Skin155.data[["Antibody Capture"]]) <- gsub("CD196-CCR6", "CD196", x = rownames(x = Skin155.data[["Antibody Capture"]]))
rownames(x = Skin155.data[["Antibody Capture"]]) <- gsub("CD195-CCR5", "CD195", x = rownames(x = Skin155.data[["Antibody Capture"]]))
rownames(x = Skin155.data[["Antibody Capture"]]) <- gsub("CD56-NCAM", "CD56", x = rownames(x = Skin155.data[["Antibody Capture"]]))
rownames(x = Skin155.data[["Antibody Capture"]]) <- gsub("[?]", "", x = rownames(x = Skin155.data[["Antibody Capture"]]))

### Skin169
rownames(x = Skin169.data[["Antibody Capture"]]) <- gsub("CD141-Thrombomodulin","CD141", x = rownames(x = Skin169.data[["Antibody Capture"]]))
rownames(x = Skin169.data[["Antibody Capture"]]) <- gsub("CD294-CRTH2", "CD294", x = rownames(x = Skin169.data[["Antibody Capture"]]))
rownames(x = Skin169.data[["Antibody Capture"]]) <- gsub("CD127-IL-7R?", "CD127", x = rownames(x = Skin169.data[["Antibody Capture"]]))
rownames(x = Skin169.data[["Antibody Capture"]]) <- gsub("CD207-Langerin", "CD207", x = rownames(x = Skin169.data[["Antibody Capture"]]))
rownames(x = Skin169.data[["Antibody Capture"]]) <- gsub("CD197-CCR7", "CD197", x = rownames(x = Skin169.data[["Antibody Capture"]]))
rownames(x = Skin169.data[["Antibody Capture"]]) <- gsub("CD196-CCR6", "CD196", x = rownames(x = Skin169.data[["Antibody Capture"]]))
rownames(x = Skin169.data[["Antibody Capture"]]) <- gsub("CD195-CCR5", "CD195", x = rownames(x = Skin169.data[["Antibody Capture"]]))
rownames(x = Skin169.data[["Antibody Capture"]]) <- gsub("CD56-NCAM", "CD56", x = rownames(x = Skin169.data[["Antibody Capture"]]))
rownames(x = Skin169.data[["Antibody Capture"]]) <- gsub("[?]", "", x = rownames(x = Skin169.data[["Antibody Capture"]]))

### Skin170
rownames(x = Skin170.data[["Antibody Capture"]]) <- gsub("CD141-Thrombomodulin","CD141", x = rownames(x = Skin170.data[["Antibody Capture"]]))
rownames(x = Skin170.data[["Antibody Capture"]]) <- gsub("CD294-CRTH2", "CD294", x = rownames(x = Skin170.data[["Antibody Capture"]]))
rownames(x = Skin170.data[["Antibody Capture"]]) <- gsub("CD127-IL-7R?", "CD127", x = rownames(x = Skin170.data[["Antibody Capture"]]))
rownames(x = Skin170.data[["Antibody Capture"]]) <- gsub("CD207-Langerin", "CD207", x = rownames(x = Skin170.data[["Antibody Capture"]]))
rownames(x = Skin170.data[["Antibody Capture"]]) <- gsub("CD197-CCR7", "CD197", x = rownames(x = Skin170.data[["Antibody Capture"]]))
rownames(x = Skin170.data[["Antibody Capture"]]) <- gsub("CD196-CCR6", "CD196", x = rownames(x = Skin170.data[["Antibody Capture"]]))
rownames(x = Skin170.data[["Antibody Capture"]]) <- gsub("CD195-CCR5", "CD195", x = rownames(x = Skin170.data[["Antibody Capture"]]))
rownames(x = Skin170.data[["Antibody Capture"]]) <- gsub("CD56-NCAM", "CD56", x = rownames(x = Skin170.data[["Antibody Capture"]]))
rownames(x = Skin170.data[["Antibody Capture"]]) <- gsub("[?]", "", x = rownames(x = Skin170.data[["Antibody Capture"]]))

### Skin165
rownames(x = Skin165.data[["Antibody Capture"]]) <- gsub("CD141-Thrombomodulin","CD141", x = rownames(x = Skin165.data[["Antibody Capture"]]))
rownames(x = Skin165.data[["Antibody Capture"]]) <- gsub("CD294-CRTH2", "CD294", x = rownames(x = Skin165.data[["Antibody Capture"]]))
rownames(x = Skin165.data[["Antibody Capture"]]) <- gsub("CD127-IL-7R?", "CD127", x = rownames(x = Skin165.data[["Antibody Capture"]]))
rownames(x = Skin165.data[["Antibody Capture"]]) <- gsub("CD207-Langerin", "CD207", x = rownames(x = Skin165.data[["Antibody Capture"]]))
rownames(x = Skin165.data[["Antibody Capture"]]) <- gsub("CD197-CCR7", "CD197", x = rownames(x = Skin165.data[["Antibody Capture"]]))
rownames(x = Skin165.data[["Antibody Capture"]]) <- gsub("CD196-CCR6", "CD196", x = rownames(x = Skin165.data[["Antibody Capture"]]))
rownames(x = Skin165.data[["Antibody Capture"]]) <- gsub("CD195-CCR5", "CD195", x = rownames(x = Skin165.data[["Antibody Capture"]]))
rownames(x = Skin165.data[["Antibody Capture"]]) <- gsub("CD56-NCAM", "CD56", x = rownames(x = Skin165.data[["Antibody Capture"]]))
rownames(x = Skin165.data[["Antibody Capture"]]) <- gsub("[?]", "", x = rownames(x = Skin165.data[["Antibody Capture"]]))

### Skin173
rownames(x = Skin173.data[["Antibody Capture"]]) <- gsub("CD141-Thrombomodulin","CD141", x = rownames(x = Skin173.data[["Antibody Capture"]]))
rownames(x = Skin173.data[["Antibody Capture"]]) <- gsub("CD294-CRTH2", "CD294", x = rownames(x = Skin173.data[["Antibody Capture"]]))
rownames(x = Skin173.data[["Antibody Capture"]]) <- gsub("CD127-IL-7R?", "CD127", x = rownames(x = Skin173.data[["Antibody Capture"]]))
rownames(x = Skin173.data[["Antibody Capture"]]) <- gsub("CD207-Langerin", "CD207", x = rownames(x = Skin173.data[["Antibody Capture"]]))
rownames(x = Skin173.data[["Antibody Capture"]]) <- gsub("CD197-CCR7", "CD197", x = rownames(x = Skin173.data[["Antibody Capture"]]))
rownames(x = Skin173.data[["Antibody Capture"]]) <- gsub("CD196-CCR6", "CD196", x = rownames(x = Skin173.data[["Antibody Capture"]]))
rownames(x = Skin173.data[["Antibody Capture"]]) <- gsub("CD195-CCR5", "CD195", x = rownames(x = Skin173.data[["Antibody Capture"]]))
rownames(x = Skin173.data[["Antibody Capture"]]) <- gsub("CD56-NCAM", "CD56", x = rownames(x = Skin165.data[["Antibody Capture"]]))
rownames(x = Skin173.data[["Antibody Capture"]]) <- gsub("[?]", "", x = rownames(x = Skin165.data[["Antibody Capture"]]))

### Skin194
rownames(x = Skin194.data[["Antibody Capture"]]) <- gsub("CD141-Thrombomodulin","CD141", x = rownames(x = Skin194.data[["Antibody Capture"]]))
rownames(x = Skin194.data[["Antibody Capture"]]) <- gsub("CD294-CRTH2", "CD294", x = rownames(x = Skin194.data[["Antibody Capture"]]))
rownames(x = Skin194.data[["Antibody Capture"]]) <- gsub("CD127-IL-7R?", "CD127", x = rownames(x = Skin194.data[["Antibody Capture"]]))
rownames(x = Skin194.data[["Antibody Capture"]]) <- gsub("CD207-Langerin", "CD207", x = rownames(x = Skin194.data[["Antibody Capture"]]))
rownames(x = Skin194.data[["Antibody Capture"]]) <- gsub("CD197-CCR7", "CD197", x = rownames(x = Skin194.data[["Antibody Capture"]]))
rownames(x = Skin194.data[["Antibody Capture"]]) <- gsub("CD196-CCR6", "CD196", x = rownames(x = Skin194.data[["Antibody Capture"]]))
rownames(x = Skin194.data[["Antibody Capture"]]) <- gsub("CD195-CCR5", "CD195", x = rownames(x = Skin194.data[["Antibody Capture"]]))
rownames(x = Skin194.data[["Antibody Capture"]]) <- gsub("CD56-NCAM", "CD56", x = rownames(x = Skin194.data[["Antibody Capture"]]))
rownames(x = Skin194.data[["Antibody Capture"]]) <- gsub("[?]", "", x = rownames(x = Skin194.data[["Antibody Capture"]]))

### Skin167
rownames(x = Skin167.data[["Antibody Capture"]]) <- gsub("CD141-Thrombomodulin","CD141", x = rownames(x = Skin167.data[["Antibody Capture"]]))
rownames(x = Skin167.data[["Antibody Capture"]]) <- gsub("CD294-CRTH2", "CD294", x = rownames(x = Skin167.data[["Antibody Capture"]]))
rownames(x = Skin167.data[["Antibody Capture"]]) <- gsub("CD127-IL-7R?", "CD127", x = rownames(x = Skin167.data[["Antibody Capture"]]))
rownames(x = Skin167.data[["Antibody Capture"]]) <- gsub("CD207-Langerin", "CD207", x = rownames(x = Skin167.data[["Antibody Capture"]]))
rownames(x = Skin167.data[["Antibody Capture"]]) <- gsub("CD197-CCR7", "CD197", x = rownames(x = Skin167.data[["Antibody Capture"]]))
rownames(x = Skin167.data[["Antibody Capture"]]) <- gsub("CD196-CCR6", "CD196", x = rownames(x = Skin167.data[["Antibody Capture"]]))
rownames(x = Skin167.data[["Antibody Capture"]]) <- gsub("CD195-CCR5", "CD195", x = rownames(x = Skin167.data[["Antibody Capture"]]))
rownames(x = Skin167.data[["Antibody Capture"]]) <- gsub("CD56-NCAM", "CD56", x = rownames(x = Skin167.data[["Antibody Capture"]]))
rownames(x = Skin167.data[["Antibody Capture"]]) <- gsub("[?]", "", x = rownames(x = Skin167.data[["Antibody Capture"]]))

### Skin192
rownames(x = Skin192.data[["Antibody Capture"]]) <- gsub("CD141-Thrombomodulin","CD141", x = rownames(x = Skin192.data[["Antibody Capture"]]))
rownames(x = Skin192.data[["Antibody Capture"]]) <- gsub("CD294-CRTH2", "CD294", x = rownames(x = Skin192.data[["Antibody Capture"]]))
rownames(x = Skin192.data[["Antibody Capture"]]) <- gsub("CD127-IL-7R?", "CD127", x = rownames(x = Skin192.data[["Antibody Capture"]]))
rownames(x = Skin192.data[["Antibody Capture"]]) <- gsub("CD207-Langerin", "CD207", x = rownames(x = Skin192.data[["Antibody Capture"]]))
rownames(x = Skin192.data[["Antibody Capture"]]) <- gsub("CD197-CCR7", "CD197", x = rownames(x = Skin192.data[["Antibody Capture"]]))
rownames(x = Skin192.data[["Antibody Capture"]]) <- gsub("CD196-CCR6", "CD196", x = rownames(x = Skin192.data[["Antibody Capture"]]))
rownames(x = Skin192.data[["Antibody Capture"]]) <- gsub("CD195-CCR5", "CD195", x = rownames(x = Skin192.data[["Antibody Capture"]]))
rownames(x = Skin192.data[["Antibody Capture"]]) <- gsub("CD56-NCAM", "CD56", x = rownames(x = Skin192.data[["Antibody Capture"]]))
rownames(x = Skin192.data[["Antibody Capture"]]) <- gsub("[?]", "", x = rownames(x = Skin192.data[["Antibody Capture"]]))
```


```{r}
####other
### Skin163
rownames(x = Skin163.data[["Antibody Capture"]]) <- gsub("CD141-Thrombomodulin","CD141", x = rownames(x = Skin163.data[["Antibody Capture"]]))
rownames(x = Skin163.data[["Antibody Capture"]]) <- gsub("CD294-CRTH2", "CD294", x = rownames(x = Skin163.data[["Antibody Capture"]]))
rownames(x = Skin163.data[["Antibody Capture"]]) <- gsub("CD127-IL-7R?", "CD127", x = rownames(x = Skin163.data[["Antibody Capture"]]))
rownames(x = Skin163.data[["Antibody Capture"]]) <- gsub("CD207-Langerin", "CD207", x = rownames(x = Skin163.data[["Antibody Capture"]]))
rownames(x = Skin163.data[["Antibody Capture"]]) <- gsub("CD197-CCR7", "CD197", x = rownames(x = Skin163.data[["Antibody Capture"]]))
rownames(x = Skin163.data[["Antibody Capture"]]) <- gsub("CD196-CCR6", "CD196", x = rownames(x = Skin163.data[["Antibody Capture"]]))
rownames(x = Skin163.data[["Antibody Capture"]]) <- gsub("CD195-CCR5", "CD195", x = rownames(x = Skin163.data[["Antibody Capture"]]))
rownames(x = Skin163.data[["Antibody Capture"]]) <- gsub("CD56-NCAM", "CD56", x = rownames(x = Skin163.data[["Antibody Capture"]]))
rownames(x = Skin163.data[["Antibody Capture"]]) <- gsub("[?]", "", x = rownames(x = Skin163.data[["Antibody Capture"]]))

### Skin175
rownames(x = Skin175.data[["Antibody Capture"]]) <- gsub("CD141-Thrombomodulin","CD141", x = rownames(x = Skin175.data[["Antibody Capture"]]))
rownames(x = Skin175.data[["Antibody Capture"]]) <- gsub("CD294-CRTH2", "CD294", x = rownames(x = Skin175.data[["Antibody Capture"]]))
rownames(x = Skin175.data[["Antibody Capture"]]) <- gsub("CD127-IL-7R?", "CD127", x = rownames(x = Skin175.data[["Antibody Capture"]]))
rownames(x = Skin175.data[["Antibody Capture"]]) <- gsub("CD207-Langerin", "CD207", x = rownames(x = Skin175.data[["Antibody Capture"]]))
rownames(x = Skin175.data[["Antibody Capture"]]) <- gsub("CD197-CCR7", "CD197", x = rownames(x = Skin175.data[["Antibody Capture"]]))
rownames(x = Skin175.data[["Antibody Capture"]]) <- gsub("CD196-CCR6", "CD196", x = rownames(x = Skin175.data[["Antibody Capture"]]))
rownames(x = Skin175.data[["Antibody Capture"]]) <- gsub("CD195-CCR5", "CD195", x = rownames(x = Skin175.data[["Antibody Capture"]]))
rownames(x = Skin175.data[["Antibody Capture"]]) <- gsub("CD56-NCAM", "CD56", x = rownames(x = Skin175.data[["Antibody Capture"]]))
rownames(x = Skin175.data[["Antibody Capture"]]) <- gsub("[?]", "", x = rownames(x = Skin175.data[["Antibody Capture"]]))

### Skin203
rownames(x = Skin203.data[["Antibody Capture"]]) <- gsub("CD141-Thrombomodulin","CD141", x = rownames(x = Skin203.data[["Antibody Capture"]]))
rownames(x = Skin203.data[["Antibody Capture"]]) <- gsub("CD294-CRTH2", "CD294", x = rownames(x = Skin203.data[["Antibody Capture"]]))
rownames(x = Skin203.data[["Antibody Capture"]]) <- gsub("CD127-IL-7R?", "CD127", x = rownames(x = Skin203.data[["Antibody Capture"]]))
rownames(x = Skin203.data[["Antibody Capture"]]) <- gsub("CD207-Langerin", "CD207", x = rownames(x = Skin203.data[["Antibody Capture"]]))
rownames(x = Skin203.data[["Antibody Capture"]]) <- gsub("CD197-CCR7", "CD197", x = rownames(x = Skin203.data[["Antibody Capture"]]))
rownames(x = Skin203.data[["Antibody Capture"]]) <- gsub("CD196-CCR6", "CD196", x = rownames(x = Skin203.data[["Antibody Capture"]]))
rownames(x = Skin203.data[["Antibody Capture"]]) <- gsub("CD195-CCR5", "CD195", x = rownames(x = Skin203.data[["Antibody Capture"]]))
rownames(x = Skin203.data[["Antibody Capture"]]) <- gsub("CD56-NCAM", "CD56", x = rownames(x = Skin203.data[["Antibody Capture"]]))
rownames(x = Skin203.data[["Antibody Capture"]]) <- gsub("[?]", "", x = rownames(x = Skin203.data[["Antibody Capture"]]))
```

double check the ADT name
```{r}
###NML
rownames(Skin150.data[["Antibody Capture"]])
rownames(Skin154.data[["Antibody Capture"]])
rownames(Skin155.data[["Antibody Capture"]])
rownames(Skin169.data[["Antibody Capture"]])
rownames(Skin195.data[["Antibody Capture"]])
rownames(Skin204.data[["Antibody Capture"]])
rownames(Skin207.data[["Antibody Capture"]])

### AD
rownames(Skin170.data[["Antibody Capture"]])
rownames(Skin198.data[["Antibody Capture"]])
rownames(Skin230.data[["Antibody Capture"]])
rownames(Skin231.data[["Antibody Capture"]])
rownames(Skin232.data[["Antibody Capture"]])
rownames(Skin233.data[["Antibody Capture"]])
rownames(Skin236.data[["Antibody Capture"]])

### Psoriasis
rownames(Skin165.data[["Antibody Capture"]])
rownames(Skin173.data[["Antibody Capture"]])
rownames(Skin194.data[["Antibody Capture"]])
rownames(Skin199.data[["Antibody Capture"]])
rownames(Skin211.data[["Antibody Capture"]])
rownames(Skin222.data[["Antibody Capture"]])
rownames(Skin234.data[["Antibody Capture"]])
rownames(Skin235.data[["Antibody Capture"]])

#####others
rownames(Skin163.data[["Antibody Capture"]])
rownames(Skin175.data[["Antibody Capture"]])
rownames(Skin203.data[["Antibody Capture"]])
```
creat seurat object with RNA data

```{r, echo=FALSE}
### normal 
Skin150 <- CreateSeuratObject(counts = Skin150.data$"Gene Expression", project = "Skin150")
Skin154 <- CreateSeuratObject(counts = Skin154.data$"Gene Expression", project = "Skin154")
Skin155 <- CreateSeuratObject(counts = Skin155.data$"Gene Expression", project = "Skin155")
Skin169 <- CreateSeuratObject(counts = Skin169.data$"Gene Expression", project = "Skin169")
Skin195 <- CreateSeuratObject(counts = Skin195.data$"Gene Expression", project = "Skin195")
Skin204 <- CreateSeuratObject(counts = Skin204.data$"Gene Expression", project = "Skin204")
Skin207 <- CreateSeuratObject(counts = Skin207.data$"Gene Expression", project = "Skin207")

### AD
Skin170 <- CreateSeuratObject(counts = Skin170.data$"Gene Expression", project = "Skin170")
Skin198 <- CreateSeuratObject(counts = Skin198.data$"Gene Expression", project = "Skin198")
Skin230 <- CreateSeuratObject(counts = Skin230.data$"Gene Expression", project = "Skin230")
Skin231 <- CreateSeuratObject(counts = Skin231.data$"Gene Expression", project = "Skin231")
Skin232 <- CreateSeuratObject(counts = Skin232.data$"Gene Expression", project = "Skin232")
Skin233 <- CreateSeuratObject(counts = Skin233.data$"Gene Expression", project = "Skin233")
Skin236 <- CreateSeuratObject(counts = Skin236.data$"Gene Expression", project = "Skin236")

### Psoriasis
Skin165 <- CreateSeuratObject(counts = Skin165.data$"Gene Expression", project = "Skin165")
Skin173 <- CreateSeuratObject(counts = Skin173.data$"Gene Expression", project = "Skin173")
Skin194 <- CreateSeuratObject(counts = Skin194.data$"Gene Expression", project = "Skin194")
Skin199 <- CreateSeuratObject(counts = Skin199.data$"Gene Expression", project = "Skin199")
Skin211 <- CreateSeuratObject(counts = Skin211.data$"Gene Expression", project = "Skin211")
Skin222 <- CreateSeuratObject(counts = Skin222.data$"Gene Expression", project = "Skin222")
Skin234 <- CreateSeuratObject(counts = Skin234.data$"Gene Expression", project = "Skin234")
Skin235 <- CreateSeuratObject(counts = Skin235.data$"Gene Expression", project = "Skin235")

### AE samples
Skin167 <- CreateSeuratObject(counts = Skin167.data$"Gene Expression", project = "Skin167")
Skin174 <- CreateSeuratObject(counts = Skin174.data, project = "Skin174")
Skin192 <- CreateSeuratObject(counts = Skin192.data$"Gene Expression", project = "Skin192")
Skin200 <- CreateSeuratObject(counts = Skin200.data$"Gene Expression", project = "Skin200")
Skin202 <- CreateSeuratObject(counts = Skin202.data$"Gene Expression", project = "Skin202")
Skin218 <- CreateSeuratObject(counts = Skin218.data$"Gene Expression", project = "Skin218")
Skin219 <- CreateSeuratObject(counts = Skin219.data$"Gene Expression", project = "Skin219")

### others
Skin141 <- CreateSeuratObject(counts = Skin141.data, project = "Skin141")
Skin163 <- CreateSeuratObject(counts = Skin163.data$"Gene Expression", project = "Skin163")
Skin175 <- CreateSeuratObject(counts = Skin175.data$"Gene Expression", project = "Skin175")
Skin203 <- CreateSeuratObject(counts = Skin203.data$"Gene Expression", project = "Skin203")
```
Change seurat object to single cell experiment to label the doublets
```{r, echo=FALSE}
### NML samples
Skin150.sce <- as.SingleCellExperiment(Skin150)
Skin154.sce <- as.SingleCellExperiment(Skin154)
Skin155.sce <- as.SingleCellExperiment(Skin155)
Skin169.sce <- as.SingleCellExperiment(Skin169)
Skin195.sce <- as.SingleCellExperiment(Skin195)
Skin204.sce <- as.SingleCellExperiment(Skin204)
Skin207.sce <- as.SingleCellExperiment(Skin207)

### AD
Skin170.sce <- as.SingleCellExperiment(Skin170)
Skin198.sce <- as.SingleCellExperiment(Skin198)
Skin230.sce <- as.SingleCellExperiment(Skin230)
Skin231.sce <- as.SingleCellExperiment(Skin231)
Skin232.sce <- as.SingleCellExperiment(Skin232)
Skin233.sce <- as.SingleCellExperiment(Skin233)
Skin236.sce <- as.SingleCellExperiment(Skin236)

### psoriasis
Skin165.sce <- as.SingleCellExperiment(Skin165)
Skin173.sce <- as.SingleCellExperiment(Skin173)
Skin194.sce <- as.SingleCellExperiment(Skin194)
Skin199.sce <- as.SingleCellExperiment(Skin199)
Skin211.sce <- as.SingleCellExperiment(Skin211)
Skin222.sce <- as.SingleCellExperiment(Skin222)
Skin234.sce <- as.SingleCellExperiment(Skin234)
Skin235.sce <- as.SingleCellExperiment(Skin235)


### AE samples
Skin167.sce <- as.SingleCellExperiment(Skin167)
Skin174.sce <- as.SingleCellExperiment(Skin174)
Skin192.sce <- as.SingleCellExperiment(Skin192)
Skin200.sce <- as.SingleCellExperiment(Skin200)
Skin202.sce <- as.SingleCellExperiment(Skin202)
Skin218.sce <- as.SingleCellExperiment(Skin218)
Skin219.sce <- as.SingleCellExperiment(Skin219)


####other
Skin141.sce <- as.SingleCellExperiment(Skin141)
Skin163.sce <- as.SingleCellExperiment(Skin163)
Skin175.sce <- as.SingleCellExperiment(Skin175)
Skin203.sce <- as.SingleCellExperiment(Skin203)
```

Check the colSums to make sure there are no empty droplet, or we will get error: Size factors should be positive
```{r, echo=FALSE}
###NML
min(colSums(counts(Skin150.sce)))
min(colSums(counts(Skin154.sce)))
min(colSums(counts(Skin155.sce)))
min(colSums(counts(Skin169.sce)))
min(colSums(counts(Skin195.sce)))
min(colSums(counts(Skin204.sce)))
min(colSums(counts(Skin207.sce)))

### AD
min(colSums(counts(Skin170.sce)))
min(colSums(counts(Skin198.sce)))
min(colSums(counts(Skin230.sce)))
min(colSums(counts(Skin231.sce)))
min(colSums(counts(Skin232.sce)))
min(colSums(counts(Skin233.sce)))
min(colSums(counts(Skin236.sce)))

### psoriasis
min(colSums(counts(Skin165.sce)))
min(colSums(counts(Skin173.sce)))
min(colSums(counts(Skin194.sce)))
min(colSums(counts(Skin199.sce))) 
min(colSums(counts(Skin211.sce)))
min(colSums(counts(Skin222.sce))) 
min(colSums(counts(Skin234.sce)))
min(colSums(counts(Skin235.sce)))


### AE samples
min(colSums(counts(Skin167.sce)))
min(colSums(counts(Skin174.sce)))
min(colSums(counts(Skin192.sce))) 
min(colSums(counts(Skin200.sce)))
min(colSums(counts(Skin202.sce))) 
min(colSums(counts(Skin218.sce)))
min(colSums(counts(Skin219.sce)))


####other
min(colSums(counts(Skin141.sce)))
min(colSums(counts(Skin163.sce))) 
min(colSums(counts(Skin175.sce)))
min(colSums(counts(Skin203.sce)))
```
We compared the doubletfinder and the scDbFinder, there is no difference between them, like cluster numbers with same resolution, cluster markers, and the cluster DEGs. The doubletfinder has many parameters associated with the doublet rate, which will affect the results. Therefore, we choose scDbFinder to label the doublet cells.
The example always run scDblFinder (providing the unsually high doublet rate), like sce <- scDblFinder(sce, dbr=0.1), but for 10x data, it is better to not set the dbr, or you will get too much higher doublets. We also tried to find doublets after integrating with sce <- scDblFinder(sce, samples="cond", BPPARAM=MulticoreParam(3)), it also leads to higher doublets, therefore, it is better to remove the doublets one sample by one sample.
```{r, echo=FALSE}
### NML samples
Skin150.sce <- scDblFinder(Skin150.sce)
Skin154.sce <- scDblFinder(Skin154.sce)
Skin155.sce <- scDblFinder(Skin155.sce)
Skin169.sce <- scDblFinder(Skin169.sce)
Skin195.sce <- scDblFinder(Skin195.sce)
Skin204.sce <- scDblFinder(Skin204.sce)
Skin207.sce <- scDblFinder(Skin207.sce)

### AD
Skin170.sce <- scDblFinder(Skin170.sce)
Skin198.sce <- scDblFinder(Skin198.sce)
Skin230.sce <- scDblFinder(Skin230.sce)
Skin231.sce <- scDblFinder(Skin231.sce)
Skin232.sce <- scDblFinder(Skin232.sce)
Skin233.sce <- scDblFinder(Skin233.sce)
Skin236.sce <- scDblFinder(Skin236.sce)

### psoriasis
Skin165.sce <- scDblFinder(Skin165.sce)
Skin173.sce <- scDblFinder(Skin173.sce)
Skin194.sce <- scDblFinder(Skin194.sce)
Skin199.sce <- scDblFinder(Skin199.sce)
Skin211.sce <- scDblFinder(Skin211.sce)
Skin222.sce <- scDblFinder(Skin222.sce)
Skin234.sce <- scDblFinder(Skin234.sce)
Skin235.sce <- scDblFinder(Skin235.sce)


### AE samples
Skin167.sce <- scDblFinder(Skin167.sce)
Skin174.sce <- scDblFinder(Skin174.sce)
Skin192.sce <- scDblFinder(Skin192.sce)
Skin200.sce <- scDblFinder(Skin200.sce)
Skin202.sce <- scDblFinder(Skin202.sce)
Skin218.sce <- scDblFinder(Skin218.sce)
Skin219.sce <- scDblFinder(Skin219.sce)

####
Skin141.sce <- scDblFinder(Skin141.sce)
Skin163.sce <- scDblFinder(Skin163.sce)
Skin175.sce <- scDblFinder(Skin175.sce)
Skin203.sce <- scDblFinder(Skin203.sce)
```

Check the doublet numbers, but do not remove them. If all the doublets are together within one cluster, then it is useful to remove them. Or it will not affect the results and will be kept.
```{r, echo=FALSE}
### NML samples
table(call=Skin150.sce$scDblFinder.class)
table(call=Skin154.sce$scDblFinder.class)
table(call=Skin155.sce$scDblFinder.class)
table(call=Skin169.sce$scDblFinder.class)
table(call=Skin195.sce$scDblFinder.class)
table(call=Skin204.sce$scDblFinder.class)
table(call=Skin207.sce$scDblFinder.class)

### AD
table(call=Skin170.sce$scDblFinder.class)
table(call=Skin198.sce$scDblFinder.class)
table(call=Skin230.sce$scDblFinder.class)
table(call=Skin231.sce$scDblFinder.class)
table(call=Skin232.sce$scDblFinder.class)
table(call=Skin233.sce$scDblFinder.class)
table(call=Skin236.sce$scDblFinder.class)

### psoriasis
table(call=Skin165.sce$scDblFinder.class)
table(call=Skin173.sce$scDblFinder.class)
table(call=Skin194.sce$scDblFinder.class)
table(call=Skin199.sce$scDblFinder.class)
table(call=Skin211.sce$scDblFinder.class)
table(call=Skin222.sce$scDblFinder.class)
table(call=Skin234.sce$scDblFinder.class)
table(call=Skin235.sce$scDblFinder.class)


### AE samples
table(call=Skin167.sce$scDblFinder.class)
table(call=Skin174.sce$scDblFinder.class)
table(call=Skin192.sce$scDblFinder.class)
table(call=Skin200.sce$scDblFinder.class)
table(call=Skin202.sce$scDblFinder.class)
table(call=Skin218.sce$scDblFinder.class)
table(call=Skin219.sce$scDblFinder.class)

######other 
table(call=Skin141.sce$scDblFinder.class)
table(call=Skin163.sce$scDblFinder.class)
table(call=Skin175.sce$scDblFinder.class)
table(call=Skin203.sce$scDblFinder.class)
```

We just label the doublets and will not remove them now. After we cluster them and check the doublets are together in one cluster with mixed cluster markers, then we will remove them. 
Covert single cell experiment back to seurat for fultering and downstream analysis
```{r, echo=FALSE}
Skin150 <- as.Seurat(Skin150.sce, project = "Skin150")
Skin154 <- as.Seurat(Skin154.sce, project = "Skin154")
Skin155 <- as.Seurat(Skin155.sce, project = "Skin155")
Skin169 <- as.Seurat(Skin169.sce, project = "Skin169")
Skin195 <- as.Seurat(Skin195.sce, project = "Skin195")
Skin204 <- as.Seurat(Skin204.sce, project = "Skin204")
Skin207 <- as.Seurat(Skin207.sce, project = "Skin207")

### AD
Skin170 <- as.Seurat(Skin170.sce, project = "Skin170")
Skin198 <- as.Seurat(Skin198.sce, project = "Skin198")
Skin230 <- as.Seurat(Skin230.sce, project = "Skin230")
Skin231 <- as.Seurat(Skin231.sce, project = "Skin231")
Skin232 <- as.Seurat(Skin232.sce, project = "Skin232")
Skin233 <- as.Seurat(Skin233.sce, project = "Skin233")
Skin236 <- as.Seurat(Skin236.sce, project = "Skin236")

### psoriasis
Skin165 <- as.Seurat(Skin165.sce, project = "Skin165")
Skin173 <- as.Seurat(Skin173.sce, project = "Skin173")
Skin194 <- as.Seurat(Skin194.sce, project = "Skin194")
Skin199 <- as.Seurat(Skin199.sce, project = "Skin199")
Skin211 <- as.Seurat(Skin211.sce, project = "Skin211")
Skin222 <- as.Seurat(Skin222.sce, project = "Skin222")
Skin234 <- as.Seurat(Skin234.sce, project = "Skin234")
Skin235 <- as.Seurat(Skin235.sce, project = "Skin235")


### AE samples
Skin167 <- as.Seurat(Skin167.sce, project = "Skin167")
Skin174 <- as.Seurat(Skin174.sce, project = "Skin174")
Skin192 <- as.Seurat(Skin192.sce, project = "Skin192")
Skin200 <- as.Seurat(Skin200.sce, project = "Skin200")
Skin202 <- as.Seurat(Skin202.sce, project = "Skin202")
Skin218 <- as.Seurat(Skin218.sce, project = "Skin218")
Skin219 <- as.Seurat(Skin219.sce, project = "Skin219")

####other samples
Skin141 <- as.Seurat(Skin141.sce, project = "Skin141")
Skin163 <- as.Seurat(Skin163.sce, project = "Skin163")
Skin175 <- as.Seurat(Skin175.sce, project = "Skin175")
Skin203 <- as.Seurat(Skin203.sce, project = "Skin203")
```

Create ADT object with ADT data
```{r, echo=FALSE}
### normal 
Skin150[["ADT"]] <- CreateAssayObject(counts = Skin150.data$"Antibody Capture")
Skin154[["ADT"]] <- CreateAssayObject(counts = Skin154.data$"Antibody Capture")
Skin155[["ADT"]] <- CreateAssayObject(counts = Skin155.data$"Antibody Capture")
Skin169[["ADT"]] <- CreateAssayObject(counts = Skin169.data$"Antibody Capture")
Skin195[["ADT"]] <- CreateAssayObject(counts = Skin195.data$"Antibody Capture")
Skin204[["ADT"]] <- CreateAssayObject(counts = Skin204.data$"Antibody Capture")
Skin207[["ADT"]] <- CreateAssayObject(counts = Skin207.data$"Antibody Capture")

### AD
Skin170[["ADT"]] <- CreateAssayObject(counts = Skin170.data$"Antibody Capture")
Skin198[["ADT"]] <- CreateAssayObject(counts = Skin198.data$"Antibody Capture")
Skin230[["ADT"]] <- CreateAssayObject(counts = Skin230.data$"Antibody Capture")
Skin231[["ADT"]] <- CreateAssayObject(counts = Skin231.data$"Antibody Capture")
Skin232[["ADT"]] <- CreateAssayObject(counts = Skin232.data$"Antibody Capture")
Skin233[["ADT"]] <- CreateAssayObject(counts = Skin233.data$"Antibody Capture")
Skin236[["ADT"]] <- CreateAssayObject(counts = Skin236.data$"Antibody Capture")

### Psoriasis
Skin165[["ADT"]] <- CreateAssayObject(counts = Skin165.data$"Antibody Capture")
Skin173[["ADT"]] <- CreateAssayObject(counts = Skin173.data$"Antibody Capture")
Skin194[["ADT"]] <- CreateAssayObject(counts = Skin194.data$"Antibody Capture")
Skin199[["ADT"]] <- CreateAssayObject(counts = Skin199.data$"Antibody Capture")
Skin211[["ADT"]] <- CreateAssayObject(counts = Skin211.data$"Antibody Capture")
Skin222[["ADT"]] <- CreateAssayObject(counts = Skin222.data$"Antibody Capture")
Skin234[["ADT"]] <- CreateAssayObject(counts = Skin234.data$"Antibody Capture")
Skin235[["ADT"]] <- CreateAssayObject(counts = Skin235.data$"Antibody Capture")

### AE samples
Skin167[["ADT"]] <- CreateAssayObject(counts = Skin167.data$"Antibody Capture")
Skin192[["ADT"]] <- CreateAssayObject(counts = Skin192.data$"Antibody Capture")
Skin200[["ADT"]] <- CreateAssayObject(counts = Skin200.data$"Antibody Capture")
Skin202[["ADT"]] <- CreateAssayObject(counts = Skin202.data$"Antibody Capture")
Skin218[["ADT"]] <- CreateAssayObject(counts = Skin218.data$"Antibody Capture")
Skin219[["ADT"]] <- CreateAssayObject(counts = Skin219.data$"Antibody Capture")


####
Skin163[["ADT"]] <- CreateAssayObject(counts = Skin163.data$"Antibody Capture")
Skin175[["ADT"]] <- CreateAssayObject(counts = Skin175.data$"Antibody Capture")
Skin203[["ADT"]] <- CreateAssayObject(counts = Skin203.data$"Antibody Capture")
```

```{r, echo=FALSE}
###Check the cell number first
### normal
table(Idents(Skin150))
table(Idents(Skin154))
table(Idents(Skin155))
table(Idents(Skin169))
table(Idents(Skin195))
table(Idents(Skin204))
table(Idents(Skin207))

### AD
table(Idents(Skin170))
table(Idents(Skin198))
table(Idents(Skin230))
table(Idents(Skin231))
table(Idents(Skin232))
table(Idents(Skin233))
table(Idents(Skin236))

### psoriasis
table(Idents(Skin165))
table(Idents(Skin173))
table(Idents(Skin194))
table(Idents(Skin199))
table(Idents(Skin211))
table(Idents(Skin222))
table(Idents(Skin234))
table(Idents(Skin235))

### AE samples
table(Idents(Skin167))
table(Idents(Skin174))
table(Idents(Skin192))
table(Idents(Skin200))
table(Idents(Skin202))
table(Idents(Skin218))
table(Idents(Skin219))

####other
table(Idents(Skin141))
table(Idents(Skin163))
table(Idents(Skin175))
table(Idents(Skin203))
```

Check mitochondrial genes for QC
```{r, echo=FALSE}
### normal
Skin150[["percent.mt"]] <- PercentageFeatureSet(Skin150, pattern = "^MT-")
Skin154[["percent.mt"]] <- PercentageFeatureSet(Skin154, pattern = "^MT-")
Skin155[["percent.mt"]] <- PercentageFeatureSet(Skin155, pattern = "^MT-")
Skin169[["percent.mt"]] <- PercentageFeatureSet(Skin169, pattern = "^MT-")
Skin195[["percent.mt"]] <- PercentageFeatureSet(Skin195, pattern = "^MT-")
Skin204[["percent.mt"]] <- PercentageFeatureSet(Skin204, pattern = "^MT-")
Skin207[["percent.mt"]] <- PercentageFeatureSet(Skin207, pattern = "^MT-")

### AD
Skin170[["percent.mt"]] <- PercentageFeatureSet(Skin170, pattern = "^MT-")
Skin198[["percent.mt"]] <- PercentageFeatureSet(Skin198, pattern = "^MT-")
Skin230[["percent.mt"]] <- PercentageFeatureSet(Skin230, pattern = "^MT-")
Skin231[["percent.mt"]] <- PercentageFeatureSet(Skin231, pattern = "^MT-")
Skin232[["percent.mt"]] <- PercentageFeatureSet(Skin232, pattern = "^MT-")
Skin233[["percent.mt"]] <- PercentageFeatureSet(Skin233, pattern = "^MT-")
Skin236[["percent.mt"]] <- PercentageFeatureSet(Skin236, pattern = "^MT-")

### psoriasis
Skin165[["percent.mt"]] <- PercentageFeatureSet(Skin165, pattern = "^MT-")
Skin173[["percent.mt"]] <- PercentageFeatureSet(Skin173, pattern = "^MT-")
Skin194[["percent.mt"]] <- PercentageFeatureSet(Skin194, pattern = "^MT-")
Skin199[["percent.mt"]] <- PercentageFeatureSet(Skin199, pattern = "^MT-")
Skin211[["percent.mt"]] <- PercentageFeatureSet(Skin211, pattern = "^MT-")
Skin222[["percent.mt"]] <- PercentageFeatureSet(Skin222, pattern = "^MT-")
Skin234[["percent.mt"]] <- PercentageFeatureSet(Skin234, pattern = "^MT-")
Skin235[["percent.mt"]] <- PercentageFeatureSet(Skin235, pattern = "^MT-")

### AE samples
Skin167[["percent.mt"]] <- PercentageFeatureSet(Skin167, pattern = "^MT-")
Skin174[["percent.mt"]] <- PercentageFeatureSet(Skin174, pattern = "^MT-")
Skin192[["percent.mt"]] <- PercentageFeatureSet(Skin192, pattern = "^MT-")
Skin200[["percent.mt"]] <- PercentageFeatureSet(Skin200, pattern = "^MT-")
Skin202[["percent.mt"]] <- PercentageFeatureSet(Skin202, pattern = "^MT-")
Skin218[["percent.mt"]] <- PercentageFeatureSet(Skin218, pattern = "^MT-")
Skin219[["percent.mt"]] <- PercentageFeatureSet(Skin219, pattern = "^MT-")

########
Skin141[["percent.mt"]] <- PercentageFeatureSet(Skin141, pattern = "^MT-")
Skin163[["percent.mt"]] <- PercentageFeatureSet(Skin163, pattern = "^MT-")
Skin175[["percent.mt"]] <- PercentageFeatureSet(Skin175, pattern = "^MT-")
Skin203[["percent.mt"]] <- PercentageFeatureSet(Skin203, pattern = "^MT-")
```

###NML-QC
Skin150
```{r, echo=FALSE}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin150, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin150')+ geom_point(shape=16, color="blue") + 
          geom_hline(yintercept=15, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 10000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin150, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin150')+ geom_point(shape=16, color="blue")+
          geom_hline(yintercept=2500, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 10000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin150$nCount_RNA, breaks = 100, main='Skin150', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=400,col="red")
plot2 <-  hist(Skin150$nFeature_RNA, breaks = 200, main='Skin150', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=250,col="red")

plot1 + plot2
```

Skin154
```{r, echo=FALSE}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin154, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin154')+ geom_point(shape=16, color="blue") + 
          geom_hline(yintercept=40, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 30000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin154, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin154')+ geom_point(shape=16, color="blue")+
          geom_hline(yintercept=4300, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 30000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin154$nCount_RNA, breaks = 100, main='Skin154', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=400,col="red")
plot2 <-  hist(Skin154$nFeature_RNA, breaks = 200, main='Skin154', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=100,col="red")

plot1 + plot2
```

Skin155
```{r, echo=FALSE}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin155, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin155')+ geom_point(shape=16, color="blue") + 
          geom_hline(yintercept=20, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 15000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin155, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin155')+ geom_point(shape=16, color="blue")+
          geom_hline(yintercept=3000, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 15000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin155$nCount_RNA, breaks = 100, main='Skin155', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=400,col="red")
plot2 <-  hist(Skin155$nFeature_RNA, breaks = 200, main='Skin155', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=200,col="red")

plot1 + plot2
```

Skin169
```{r, echo=FALSE}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin169, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin169')+ geom_point(shape=16, color="blue") + 
          geom_hline(yintercept=40, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin169, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin169')+ geom_point(shape=16, color="blue")+
          geom_hline(yintercept=4000, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin169$nCount_RNA, breaks = 100, main='Skin169', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=400,col="red")
plot2 <-  hist(Skin169$nFeature_RNA, breaks = 200, main='Skin169', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=0,col="red")

plot1 + plot2
```

Skin195
```{r, echo=FALSE}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin195, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin195')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=40, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 30000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin195, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin195')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=4500, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 30000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin195$nCount_RNA, breaks = 100, main='Skin195', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=100,col="red")
plot2 <-  hist(Skin195$nFeature_RNA, breaks = 200, main='Skin195', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=100,col="red")

plot1 + plot2
```
Skin204
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin204, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin204')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=50, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin204, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin204')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=3500, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin204$nCount_RNA, breaks = 100, main='Skin204', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=100,col="red")
plot2 <-  hist(Skin204$nFeature_RNA, breaks = 200, main='Skin204', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=100,col="red")

plot1 + plot2
```

Skin207
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin207, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin207')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=50, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 25000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin207, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin207')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=4500, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 25000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin207$nCount_RNA, breaks = 100, main='Skin207', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=0,col="red")
plot2 <-  hist(Skin207$nFeature_RNA, breaks = 200, main='Skin207', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=100,col="red")

plot1 + plot2
```
###psoriasis-QC
Skin165
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin165, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin165')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=25, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin165, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin165')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=4000, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin165$nCount_RNA, breaks = 100, main='Skin165', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=400,col="red")
plot2 <-  hist(Skin165$nFeature_RNA, breaks = 200, main='Skin165', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=200,col="red")

plot1 + plot2
```

Skin173
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin173, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin173')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=25, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin173, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin173')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=4000, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin173$nCount_RNA, breaks = 100, main='Skin173', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=400,col="red")
plot2 <-  hist(Skin173$nFeature_RNA, breaks = 200, main='Skin173', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=100,col="red")

plot1 + plot2
```

Skin194
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin194, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin194')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=40, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 23000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin194, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin194')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=4500, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 23000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin194$nCount_RNA, breaks = 100, main='Skin194', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=400,col="red")
plot2 <-  hist(Skin194$nFeature_RNA, breaks = 200, main='Skin194', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=250,col="red")

plot1 + plot2
```
Skin199
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin199, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin199')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=40, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin199, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin199')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=4000, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin199$nCount_RNA, breaks = 100, main='Skin199', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=100,col="red")
plot2 <-  hist(Skin199$nFeature_RNA, breaks = 200, main='Skin199', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=100,col="red")

plot1 + plot2
```
Skin211

```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin211, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin211')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=40, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin211, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin211')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=3500, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin211$nCount_RNA, breaks = 100, main='Skin211', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=0,col="red")
plot2 <-  hist(Skin211$nFeature_RNA, breaks = 200, main='Skin211', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=100,col="red")

plot1 + plot2
```

Skin222
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin222, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin222')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=25, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin222, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin222')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=4000, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin222$nCount_RNA, breaks = 100, main='Skin222', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=100,col="red")
plot2 <-  hist(Skin222$nFeature_RNA, breaks = 200, main='Skin222', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=100,col="red")

plot1 + plot2
```

Skin234
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin234, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin234')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=30, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 15000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin234, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin234')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=4000, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 15000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin234$nCount_RNA, breaks = 100, main='Skin234', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=100,col="red")
plot2 <-  hist(Skin234$nFeature_RNA, breaks = 200, main='Skin234', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=100,col="red")

plot1 + plot2
```

Skin235
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin235, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin235')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=40, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 30000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin235, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin235')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=4500, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 30000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin235$nCount_RNA, breaks = 100, main='Skin235', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=0,col="red")
plot2 <-  hist(Skin235$nFeature_RNA, breaks = 200, main='Skin235', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=100,col="red")

plot1 + plot2
```
###AE-QC
Skin167
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin167, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin167')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=25, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 25000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin167, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin167')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=4500, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 25000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin167$nCount_RNA, breaks = 100, main='Skin167', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=400,col="red")
plot2 <-  hist(Skin167$nFeature_RNA, breaks = 200, main='Skin167', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=200,col="red")

plot1 + plot2
```

Skin174
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin174, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin174')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=10, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 15000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin174, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin174')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=4000, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 15000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin174$nCount_RNA, breaks = 100, main='Skin174', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=400,col="red")
plot2 <-  hist(Skin174$nFeature_RNA, breaks = 200, main='Skin174', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=200,col="red")

plot1 + plot2
```

Skin192
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin192, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin192')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=40, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 30000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin192, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin192')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=4500, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 30000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin192$nCount_RNA, breaks = 100, main='Skin192', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=500,col="red")
plot2 <-  hist(Skin192$nFeature_RNA, breaks = 200, main='Skin192', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=0,col="red")

plot1 + plot2
```

Skin200
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin200, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin200')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=40, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin200, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin200')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=4000, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin200$nCount_RNA, breaks = 100, main='Skin200', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=100,col="red")
plot2 <-  hist(Skin200$nFeature_RNA, breaks = 200, main='Skin200', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=100,col="red")

plot1 + plot2
```

Skin202
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin202, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin202')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=40, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 25000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin202, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin202')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=4000, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 25000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin202$nCount_RNA, breaks = 100, main='Skin202', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=100,col="red")
plot2 <-  hist(Skin202$nFeature_RNA, breaks = 200, main='Skin202', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=150,col="red")

plot1 + plot2
```

Skin218
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin218, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin218')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=40, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 25000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin218, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin218')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=4000, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 25000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin218$nCount_RNA, breaks = 100, main='Skin218', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=100,col="red")
plot2 <-  hist(Skin218$nFeature_RNA, breaks = 200, main='Skin218', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=100,col="red")

plot1 + plot2
```

Skin219
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin219, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin219')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=40, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 25000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin219, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin219')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=4000, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 25000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin219$nCount_RNA, breaks = 100, main='Skin219', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=100,col="red")
plot2 <-  hist(Skin219$nFeature_RNA, breaks = 200, main='Skin219', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=100,col="red")

plot1 + plot2
```
### AD-QC
Skin170
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin170, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin170')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=30, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin170, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin170')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=5000, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin170$nCount_RNA, breaks = 100, main='Skin170', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=400,col="red")
plot2 <-  hist(Skin170$nFeature_RNA, breaks = 200, main='Skin170', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=200,col="red")

plot1 + plot2
```

Skin198
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin198, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin198')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=25, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin198, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin198')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=4000, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin198$nCount_RNA, breaks = 100, main='Skin198', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=100,col="red")
plot2 <-  hist(Skin198$nFeature_RNA, breaks = 200, main='Skin198', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=200,col="red")

plot1 + plot2
```

Skin230
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin230, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin230')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=40, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin230, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin230')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=4000, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 25000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin230$nCount_RNA, breaks = 100, main='Skin230', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=0,col="red")
plot2 <-  hist(Skin230$nFeature_RNA, breaks = 200, main='Skin230', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=100,col="red")

plot1 + plot2
```

Skin231
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin231, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin231')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=30, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin231, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin231')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=3500, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin231$nCount_RNA, breaks = 100, main='Skin231', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=100,col="red")
plot2 <-  hist(Skin231$nFeature_RNA, breaks = 200, main='Skin231', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=100,col="red")

plot1 + plot2
```

Skin232
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin232, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin232')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=30, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin232, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin232')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=4000, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin232$nCount_RNA, breaks = 100, main='Skin232', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=100,col="red")
plot2 <-  hist(Skin232$nFeature_RNA, breaks = 200, main='Skin232', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=100,col="red")

plot1 + plot2
```

Skin233
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin233, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin233')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=25, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin233, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin233')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=4000, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 20000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin233$nCount_RNA, breaks = 100, main='Skin233', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=100,col="red")
plot2 <-  hist(Skin233$nFeature_RNA, breaks = 200, main='Skin233', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=200,col="red")

plot1 + plot2
```

Skin236
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin236, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin236')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=40, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 40000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin236, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin236')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=6000, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 40000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin236$nCount_RNA, breaks = 100, main='Skin236', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=100,col="red")
plot2 <-  hist(Skin236$nFeature_RNA, breaks = 200, main='Skin236', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=100,col="red")

plot1 + plot2
```
Skin141
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin141, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin141')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=40, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 40000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin141, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin141')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=6000, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 40000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin141$nCount_RNA, breaks = 100, main='Skin141', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=100,col="red")
plot2 <-  hist(Skin141$nFeature_RNA, breaks = 200, main='Skin141', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=100,col="red")

plot1 + plot2
```

Skin163
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin163, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin163')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=40, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 40000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin163, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin163')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=6000, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 40000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin163$nCount_RNA, breaks = 100, main='Skin163', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=100,col="red")
plot2 <-  hist(Skin163$nFeature_RNA, breaks = 200, main='Skin163', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=100,col="red")

plot1 + plot2
```

Skin175
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin175, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin175')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=40, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 40000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin175, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin175')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=6000, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 40000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin175$nCount_RNA, breaks = 100, main='Skin175', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=100,col="red")
plot2 <-  hist(Skin175$nFeature_RNA, breaks = 200, main='Skin175', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=100,col="red")

plot1 + plot2
```

Skin203
```{r}
###scatterplot
options(repr.plot.height = 6, repr.plot.width = 15)
p1 <- FeatureScatter(Skin203, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle('Skin203')+ geom_point(shape=16, color="blue") + 
  geom_hline(yintercept=40, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 40000, linetype="dashed", color = "black", size=1.5)

p2 <- FeatureScatter(Skin203, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle('Skin203')+ geom_point(shape=16, color="blue")+
  geom_hline(yintercept=6000, linetype="dashed", color = "black", size=1.5) + geom_vline(xintercept = 40000, linetype="dashed", color = "black", size=1.5)

p1+p2

### histograph
options(repr.plot.height = 6, repr.plot.width = 12)
plot1 <-  hist(Skin203$nCount_RNA, breaks = 100, main='Skin203', xlab = "Count depth (Counts/cell)", col='wheat')
plot1 <-  abline(v=100,col="red")
plot2 <-  hist(Skin203$nFeature_RNA, breaks = 200, main='Skin203', xlab = "Number of genes per cell", xlim = c(0, 4000), col='wheat')
plot2 <-  abline(v=100,col="red")

plot1 + plot2
```

Set the cutoff and filter the low-quality cells according to the upper scatter plot. We will use the uniform cutoff for all the samples. Then it will be easier to write the Method part.
```{r}
### normal
Skin150 <- subset(Skin150, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin154 <- subset(Skin154, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin155 <- subset(Skin155, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin169 <- subset(Skin169, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin195 <- subset(Skin195, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin204 <- subset(Skin204, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin207 <- subset(Skin207, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)

### AD
Skin170 <- subset(Skin170, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin198 <- subset(Skin198, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin230 <- subset(Skin230, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin231 <- subset(Skin231, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin232 <- subset(Skin232, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin233 <- subset(Skin233, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin236 <- subset(Skin236, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)

### psoriasis
Skin165 <- subset(Skin165, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin173 <- subset(Skin173, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin194 <- subset(Skin194, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin199 <- subset(Skin199, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin211 <- subset(Skin211, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin222 <- subset(Skin222, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin234 <- subset(Skin234, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin235 <- subset(Skin235, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)


### AE samples
Skin167 <- subset(Skin167, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin174 <- subset(Skin174, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin192 <- subset(Skin192, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin200 <- subset(Skin200, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin202 <- subset(Skin202, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin218 <- subset(Skin218, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin219 <- subset(Skin219, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)

####other
Skin141 <- subset(Skin141, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin163 <- subset(Skin163, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin175 <- subset(Skin175, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
Skin203 <- subset(Skin203, subset = nFeature_RNA > 100 & nFeature_RNA < 6000 & percent.mt < 20)
```

Check cell numbers after QC
```{r}
### normal
table(Idents(Skin150))
table(Idents(Skin154))
table(Idents(Skin155))
table(Idents(Skin169))
table(Idents(Skin195))
table(Idents(Skin204))
table(Idents(Skin207))

### AD
table(Idents(Skin170))
table(Idents(Skin198))
table(Idents(Skin230))
table(Idents(Skin231))
table(Idents(Skin232))
table(Idents(Skin233))
table(Idents(Skin236))

### psoriasis
table(Idents(Skin165))
table(Idents(Skin173))
table(Idents(Skin194))
table(Idents(Skin199))
table(Idents(Skin211))
table(Idents(Skin222))
table(Idents(Skin234))
table(Idents(Skin235))

### AE samples
table(Idents(Skin167))
table(Idents(Skin174))
table(Idents(Skin192))
table(Idents(Skin200))
table(Idents(Skin202))
table(Idents(Skin218))
table(Idents(Skin219))

######other
table(Idents(Skin141))
table(Idents(Skin163))
table(Idents(Skin175))
table(Idents(Skin203))
```
###Assign condition for batch correction
First, we assign the donor number for each sample, which is the most important heterogenity
```{r}
### individual sample
###NML
Skin150$donor <- "150"
Skin154$donor <- "154"
Skin155$donor <- "155"
Skin169$donor <- "169"
Skin195$donor <- "195"
Skin204$donor <- "204"
Skin207$donor <- "207"

### AD
Skin170$donor <- "170"
Skin198$donor <- "198"
Skin230$donor <- "230"
Skin231$donor <- "231"
Skin232$donor <- "232"
Skin233$donor <- "233"
Skin236$donor <- "236"

### psoriasis
Skin165$donor <- "165"
Skin173$donor <- "173"
Skin194$donor <- "194"
Skin199$donor <- "199"
Skin211$donor <- "211"
Skin222$donor <- "222"
Skin234$donor <- "234"
Skin235$donor <- "235"


### AE samples
Skin167$donor <- "167"
Skin174$donor <- "174"
Skin192$donor <- "192"
Skin200$donor <- "200"
Skin202$donor <- "202"
Skin218$donor <- "218"
Skin219$donor <- "219"

####other
Skin141$donor <- "141"
Skin163$donor <- "163"
Skin175$donor <- "175"
Skin203$donor <- "203"
```

Then we assign the disease status, but this one is not for batch correction
```{r}
#### disease
### Normal
Skin150$dis <- "N"
Skin154$dis <- "N"
Skin155$dis <- "N"
Skin169$dis <- "N"
Skin195$dis <- "N"
Skin204$dis <- "N"
Skin207$dis <- "N"

### AD
Skin170$dis <- "AD"
Skin198$dis <- "AD"
Skin230$dis <- "AD"
Skin231$dis <- "AD"
Skin232$dis <- "AD"
Skin233$dis <- "AD"
Skin236$dis <- "AD"

### psoriasis
Skin165$dis <- "Pso"
Skin173$dis <- "Pso"
Skin194$dis <- "Pso"
Skin199$dis <- "Pso"
Skin211$dis <- "Pso"
Skin222$dis <- "Pso"
Skin234$dis <- "Pso"
Skin235$dis <- "Pso"


### AE samples
Skin167$dis <- "AE"
Skin174$dis <- "AE"
Skin192$dis <- "AE"
Skin200$dis <- "AE"
Skin202$dis <- "AE"
Skin218$dis <- "AE"
Skin219$dis <- "AE"

####other
Skin141$dis <- "LP"
Skin163$dis <- "LP"
Skin175$dis <- "BP"
Skin203$dis <- "MC"
```

There are some patients with treatment when collecting the samples, we will also lable this so that we can compare some signals.
```{r}
#### treat
### Normal
Skin150$treat <- "No"
Skin154$treat <- "No"
Skin155$treat <- "No"
Skin169$treat <- "No"
Skin195$treat <- "No"
Skin204$treat <- "No"
Skin207$treat <- "No"

### AD
Skin170$treat <- "Yes"
Skin198$treat <- "Yes"
Skin230$treat <- "No"
Skin231$treat <- "No"
Skin232$treat <- "No"
Skin233$treat <- "No"
Skin236$treat <- "No"

### psoriasis
Skin165$treat <- "No"
Skin173$treat <- "No"
Skin194$treat <- "No"
Skin199$treat <- "Yes"
Skin211$treat <- "Yes"
Skin222$treat <- "No"
Skin234$treat <- "No"
Skin235$treat <- "No"


### AE samples
Skin167$treat <- "No"
Skin174$treat <- "No"
Skin192$treat <- "Yes"
Skin200$treat <- "Yes"
Skin202$treat <- "Yes"
Skin218$treat <- "No"
Skin219$treat <- "No"

####Other
Skin141$treat <- "No"
Skin163$treat <- "No"
Skin175$treat <- "No"
Skin203$treat <- "Yes"
```

There are some samples using the 10x v2 chemistry while others are 10x v3, which is the techonoly batch effect and we will correct it.
```{r}
#### 10x chemistry version
### Normal
Skin150$chem <- "v2"
Skin154$chem <- "v2"
Skin155$chem <- "v2"
Skin169$chem <- "v3"
Skin195$chem <- "v3"
Skin204$chem <- "v3"
Skin207$chem <- "v3"

### AD
Skin170$chem <- "v3"
Skin198$chem <- "v3"
Skin230$chem <- "v3"
Skin231$chem <- "v3"
Skin232$chem <- "v3"
Skin233$chem <- "v3"
Skin236$chem <- "v3"

### psoriasis
Skin165$chem <- "v3"
Skin173$chem <- "v3"
Skin194$chem <- "v3"
Skin199$chem <- "v3"
Skin211$chem <- "v3"
Skin222$chem <- "v3"
Skin234$chem <- "v3"
Skin235$chem <- "v3"


### AE samples
Skin167$chem <- "v3"
Skin174$chem <- "v3"
Skin192$chem <- "v3"
Skin200$chem <- "v3"
Skin202$chem <- "v3"
Skin218$chem <- "v3"
Skin219$chem <- "v3"

### AE samples
Skin141$chem <- "v2"
Skin163$chem <- "v3"
Skin175$chem <- "v3"
Skin203$chem <- "v3"
```

There is also the order difference for the CITE-seq antibodies, even though we all collected the CD45+ cells. We don't want this batch effect in the downstream analysis
```{r}
#### pre- or post- flow
### Normal
Skin150$order <- "pre"
Skin154$order <- "post"
Skin155$order <- "post"
Skin169$order <- "post"
Skin195$order <- "post"
Skin204$order <- "post"
Skin207$order <- "post"

### AD
Skin170$order <- "pre"
Skin198$order <- "post"
Skin230$order <- "post"
Skin231$order <- "post"
Skin232$order <- "post"
Skin233$order <- "post"
Skin236$order <- "post"

### psoriasis
Skin165$order <- "pre"
Skin173$order <- "pre"
Skin194$order <- "post"
Skin199$order <- "post"
Skin211$order <- "post"
Skin222$order <- "post"
Skin234$order <- "post"
Skin235$order <- "post"


### AE samples
Skin167$order <- "pre"
Skin174$order <- "pre"
Skin192$order <- "pre"
Skin200$order <- "post"
Skin202$order <- "post"
Skin218$order <- "post"
Skin219$order <- "post"

####other
Skin141$order <- "NA"
Skin163$order <- "pre"
Skin175$order <- "pre"
Skin203$order <- "post"
```

We also have spike in for some samples in order to exclude the background from ADT data. We wil also batch correct them.
```{r}
#### mouse spike-in genes
### Normal
Skin150$spike <- "No"
Skin154$spike <- "No"
Skin155$spike <- "No"
Skin169$spike <- "No"
Skin195$spike <- "yes"
Skin204$spike <- "yes"
Skin207$spike <- "yes"

### AD
Skin170$spike <- "No"
Skin198$spike <- "yes"
Skin230$spike <- "yes"
Skin231$spike <- "yes"
Skin232$spike <- "yes"
Skin233$spike <- "yes"
Skin236$spike <- "yes"

### psoriasis
Skin165$spike <- "No"
Skin173$spike <- "No"
Skin194$spike <- "No"
Skin199$spike <- "yes"
Skin211$spike <- "yes"
Skin222$spike <- "yes"
Skin234$spike <- "yes"
Skin235$spike <- "yes"


### AE samples
Skin167$spike <- "No"
Skin174$spike <- "No"
Skin192$spike <- "No"
Skin200$spike <- "yes"
Skin202$spike <- "yes"
Skin218$spike <- "yes"
Skin219$spike <- "yes"

### AE samples
Skin141$spike <- "No"
Skin163$spike <- "No"
Skin175$spike <- "No"
Skin203$spike <- "yes"
```

### Now will work on the integration, batch correction, and  custring.
Merge all data together
```{r}
Merge <- merge(x = Skin150, y = c(Skin154, Skin155, Skin169, Skin195, Skin204, Skin207, Skin170, Skin198, Skin230, Skin231, Skin232, Skin233, Skin236, Skin165, Skin173, Skin194, Skin199, Skin211, Skin222, Skin234, Skin235, Skin167, Skin174, Skin192, Skin200, Skin202, Skin218, Skin219, Skin141, Skin163,Skin175,Skin203))
```

We will work on the ADT and RNA data integration, which is acceptable for Seurat 1.0

For RNA normalization, we will use RC normalization, as there is substantial imbalances in the analysis of log-normalized data. In particular, cells with low total UMI counts exhibited disproportionately higher variance for high-abundance genes, dampening the variance contribution from other gene groups . After testing an alternative to log-normalization (“relative counts” normalization), where we simply divided counts by total sequencing depth. Removing the log-transformation mitigated the relationships between gene expression, gene variance, and sequencing depth, but residual effects remained in both cases.

```{r}
DefaultAssay(Merge) <- 'RNA'
Merge <- NormalizeData(Merge, verbose = T, normalization.method = "LogNormalize") %>% FindVariableFeatures(selection.method = "vst", nfeatures = 2000) %>% ScaleData() %>% RunPCA()
```

we will use all ADT features for dimensional reduction
we set a dimensional reduction name to avoid overwriting the RNA dimension
```{r}
DefaultAssay(Merge) <- 'ADT'
VariableFeatures(Merge) <- rownames(Merge[["ADT"]])
Merge <- NormalizeData(Merge, normalization.method = 'CLR', margin = 2) %>% 
  ScaleData() %>% RunPCA(reduction.name = 'apca')
```
We will calculate the cell cycle score just in case we will compare the cell status. But we will not regress the cell cycle, as we may analyze the cell cycle in different diseases.
```{r}
DefaultAssay(Merge) <- 'RNA'

s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes

Merge <- CellCycleScoring(Merge, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)

VlnPlot(Merge, features = c("S.Score","G2M.Score"))
```
We will check the cell PCA before harmony so that we can compare the batch correction results.
```{r}
options(repr.plot.height = 5, repr.plot.width = 12)
p1 <- DimPlot(object = Merge, reduction = "pca", pt.size = .1, group.by = "donor")
p2 <- VlnPlot(object = Merge, features = "PC_1", group.by = "donor", pt.size = .1)+NoLegend()
plot_grid(p1,p2)

options(repr.plot.height = 5, repr.plot.width = 12)
p1 <- DimPlot(object = Merge, reduction = "pca", pt.size = .1, group.by = "order")
p2 <- VlnPlot(object = Merge, features = "PC_1", group.by = "order", pt.size = .1)+NoLegend()
plot_grid(p1,p2)

options(repr.plot.height = 5, repr.plot.width = 12)
p1 <- DimPlot(object = Merge, reduction = "pca", pt.size = .1, group.by = "chem")
p2 <- VlnPlot(object = Merge, features = "PC_1", group.by = "chem", pt.size = .1)+NoLegend()
plot_grid(p1,p2)

options(repr.plot.height = 5, repr.plot.width = 12)
p1 <- DimPlot(object = Merge, reduction = "pca", pt.size = .1, group.by = "spike")
p2 <- VlnPlot(object = Merge, features = "PC_1", group.by = "spike", pt.size = .1)+NoLegend()
plot_grid(p1,p2)
```

Now we will run harmony for batch correction. As mentioned above, we will exclude the factors we don't want to consider, such as the individual donor, flow order, different chemisty, and mouse cell spike-in or not.
```{r}
library(harmony)
options(repr.plot.height = 6, repr.plot.width = 10)
Merge <- Merge %>% 
    RunHarmony(c("donor", "order", "chem", "spike"), plot_convergence = TRUE, max.iter.harmony = 20)
```

Check the plots after harmony. The plot is like a line is becasue the mouse cells. But it doesn't matter as we found all the mouse cells are together while the remaining cell can group well.
```{r}
options(repr.plot.height = 5, repr.plot.width = 12)
p1 <- DimPlot(object = Merge, reduction = "harmony", pt.size = .1, group.by = "donor")
p2 <- VlnPlot(object = Merge, features = "harmony_1", group.by = "donor", pt.size = .1)+NoLegend()
plot_grid(p1,p2)

options(repr.plot.height = 5, repr.plot.width = 12)
p1 <- DimPlot(object = Merge, reduction = "harmony", pt.size = .1, group.by = "order")
p2 <- VlnPlot(object = Merge, features = "harmony_1", group.by = "order", pt.size = .1)+NoLegend()
plot_grid(p1,p2)

options(repr.plot.height = 5, repr.plot.width = 12)
p1 <- DimPlot(object = Merge, reduction = "harmony", pt.size = .1, group.by = "chem")
p2 <- VlnPlot(object = Merge, features = "harmony_1", group.by = "chem", pt.size = .1)+NoLegend()
plot_grid(p1,p2)

options(repr.plot.height = 5, repr.plot.width = 12)
p1 <- DimPlot(object = Merge, reduction = "harmony", pt.size = .1, group.by = "spike")
p2 <- VlnPlot(object = Merge, features = "harmony_1", group.by = "spike", pt.size = .1)+NoLegend()
plot_grid(p1,p2)
```

### We choose harmony rather than pca in FindMultiModalNeighbors is because harmony is based on PCA with bath correction, while PCA has no batch correction.

For each cell, we calculate its closest neighbors in the dataset based on a weighted combination of RNA and protein similarities. The cell-specific modality weights and multimodal neighbors are calculated in a single function, which takes ~2 minutes to run on this dataset. We specify the dimensionality of each modality (similar to specifying the number of PCs to include in scRNA-seq clustering), but you can vary these settings to see that small changes have minimal effect on the overall results.
```{r}
# Identify multimodal neighbors. These will be stored in the neighbors slot, and can be accessed using Merge[['weighted.nn']]. The WNN graph can be accessed at Merge[["wknn"]], and the SNN graph used for clustering at Merge[["wsnn"]]. Cell-specific modality weights can be accessed at Merge$RNA.weight

Merge <- FindMultiModalNeighbors(
  Merge, reduction.list = list("harmony", "apca"), 
  dims.list = list(1:30, 1:20), modality.weight.name = "RNA.weight"
)
```

```{r}
ElbowPlot(Merge, ndims = 30)
```


```{r}
options(repr.plot.height =20, repr.plot.width = 15)
DimHeatmap(Merge, dims = 1:15, cells = 500, balanced = TRUE)
```


```{r}
options(repr.plot.height =20, repr.plot.width = 15)
DimHeatmap(Merge, dims = 16:30, cells = 500, balanced = TRUE)
```

We can now use these results for downstream analysis, such as visualization and clustering. For example, we can create a UMAP visualization of the data based on a weighted combination of RNA and protein data We can also perform graph-based clustering and visualize these results on the UMAP, alongside a set of cell annotations.

```{r}
Merge <- Merge %>% 
    RunUMAP(nn.name = "weighted.nn", reduction.name = "wnn.umap", reduction.key = "wnnUMAP_") %>% 
    RunTSNE(nn.name = "weighted.nn", reduction.name = "wnn.tsne", reduction.key = "wnnTSNE_") %>% 
    FindClusters(graph.name = "wsnn", algorithm = 1, resolution = 0.1, verbose = T)
```


```{r}
DimPlot(Merge, reduction = 'wnn.umap', label = TRUE, repel = TRUE, label.size = 2.5) + NoLegend()
```
```{r}
DimPlot(Merge, reduction = 'wnn.umap', label = TRUE, repel = TRUE, label.size = 2.5, split.by = "dis") + NoLegend()
```


```{r}
Merge <- RunUMAP(Merge, reduction = 'harmony', dims = 1:18, assay = 'RNA', 
              reduction.name = 'rna.umap', reduction.key = 'rnaUMAP_')
Merge <- RunUMAP(Merge, reduction = 'apca', dims = 1:20, assay = 'ADT', 
              reduction.name = 'adt.umap', reduction.key = 'adtUMAP_')

p1 <- DimPlot(Merge, reduction = 'rna.umap', label = TRUE, repel = TRUE, label.size = 2.5) + NoLegend()
p2 <- DimPlot(Merge, reduction = 'adt.umap', label = TRUE, repel = TRUE, label.size = 2.5) + NoLegend()
p1 + p2
```
```{r}
DimPlot(Merge, reduction = 'rna.umap', label = TRUE, repel = TRUE, label.size = 2.5, split.by = "dis") + NoLegend()
```


```{r}
Merge <- Merge %>% 
    RunUMAP(reduction = "harmony", dims = 1:18) %>% 
    RunTSNE(reduction = "harmony", dims = 1:18) %>% 
    FindNeighbors(reduction = "harmony", dims = 1:18)%>%
    FindClusters(resolution = 0.1, algorithm = 1)
```


```{r}
options(repr.plot.height = 8, repr.plot.width = 24)
p1 <- DimPlot(Merge, reduction = "umap",  pt.size = .1, group.by = "donor")  + NoLegend()
p2 <- DimPlot(Merge, reduction = "umap",  pt.size = .1, group.by = "dis")
p3 <- DimPlot(Merge, reduction = "umap",  pt.size = .1, label = T) + NoLegend()
plot_grid(p1, p2, p3, ncol=3)
```

```{r}
saveRDS(Merge, "E:/Human - 10 clusters/yale-method/0208-2021-Allsamples/Skin_AllDis.rds")
```

```{r}
library(clustree)

Merge <- Merge %>%
  FindClusters(graph.name = "RNA_snn", algorithm = 1, resolution = 0.1, verbose = T)%>%
  FindClusters(graph.name = "RNA_snn", algorithm = 1, resolution = 0.2, verbose = T)%>%
  FindClusters(graph.name = "RNA_snn", algorithm = 1, resolution = 0.3, verbose = T)%>%
  FindClusters(graph.name = "RNA_snn", algorithm = 1, resolution = 0.4, verbose = T)%>%
  FindClusters(graph.name = "RNA_snn", algorithm = 1, resolution = 0.5, verbose = T)%>%
  FindClusters(graph.name = "RNA_snn", algorithm = 1, resolution = 0.6, verbose = T)%>%
  FindClusters(graph.name = "RNA_snn", algorithm = 1, resolution = 0.7, verbose = T)%>%
  FindClusters(graph.name = "RNA_snn", algorithm = 1, resolution = 0.8, verbose = T)%>%
  FindClusters(graph.name = "RNA_snn", algorithm = 1, resolution = 0.9, verbose = T)%>%
  FindClusters(graph.name = "RNA_snn", algorithm = 1, resolution = 1, verbose = T)%>%
  FindClusters(graph.name = "RNA_snn", algorithm = 1, resolution = 1.1, verbose = T)%>%
  FindClusters(graph.name = "RNA_snn", algorithm = 1, resolution = 1.2, verbose = T)%>%
  FindClusters(graph.name = "RNA_snn", algorithm = 1, resolution = 1.3, verbose = T)%>%
  FindClusters(graph.name = "RNA_snn", algorithm = 1, resolution = 1.4, verbose = T)%>%
  FindClusters(graph.name = "RNA_snn", algorithm = 1, resolution = 1.5, verbose = T)%>%
  FindClusters(graph.name = "RNA_snn", algorithm = 1, resolution = 1.6, verbose = T)%>%
  FindClusters(graph.name = "RNA_snn", algorithm = 1, resolution = 1.7, verbose = T)%>%
  FindClusters(graph.name = "RNA_snn", algorithm = 1, resolution = 1.8, verbose = T)%>%
  FindClusters(graph.name = "RNA_snn", algorithm = 1, resolution = 1.9, verbose = T)%>%
  FindClusters(graph.name = "RNA_snn", algorithm = 1, resolution = 2, verbose = T)

clustree(Merge)
```

```{r}
Merge <- Merge %>%
  FindClusters(graph.name = "RNA_snn", algorithm = 1, resolution = 0.5, verbose = T)
```
```{r}
options(repr.plot.height =5, repr.plot.width = 10)
p2 <- DimPlot(Merge, reduction = "umap",  pt.size = .1, group.by = "dis")
p3 <- DimPlot(Merge, reduction = "umap",  pt.size = .1, label = T) + NoLegend()
plot_grid(p2, p3, ncol=3)
```


```{r}
Merge <- Merge %>%
  FindClusters(graph.name = "RNA_snn", algorithm = 1, resolution = 0.6, verbose = T)
```

```{r}
DimPlot(Merge, reduction = "umap",  pt.size = .1, label = T) + NoLegend()
```

```{r}
DefaultAssay(Merge) <-"RNA"
marker_i <- FindAllMarkers(Merge, verbose = TRUE, assay = "RNA", slot = "data", logfc.threshold = 0.25)
write.xlsx(marker_i, file = "E:/Human - 10 clusters/yale-method/0208-2021-Allsamples/FindAllMarker-27Clusters.xlsx")

```

